src/dm-area/dm-map.js

```diff
@@ -696,6 +700,59 @@ export const DmMap = ({
     mapContext.drawImage(fogCanvasRef.current, 0, 0, width, height);
   };
 
+  const sendMap = useCallback(async () => {
```

Comment:
This file is getting big, I have no problem with it.

Just when files start getting large, might be time to think about splitting functionality up. Benefits it will be easier for new people to understand and maintain.